packages/marshal/test/encodePassable.test.js

```diff
@@ -353,31 +337,58 @@ test('compact custom encoding validity constraints', t => {
   }
 });
 
-const orderInvariants = (x, y) => {
+// TODO: Make a common utility for finding the first difference between iterables.
+// import { firstDiff } from '...';
+// const firstStringDiff = (a, b) => firstDiff(a, b, { getIncrement: ch => ch.length });
+// const commonStringPrefix = (a, b) => a.substring(0, firstStringDiff(a, b).index);
+const commonStringPrefix = (str1, str2) => {
+  const iter1 = str1[Symbol.iterator]();
+  const iter2 = str2[Symbol.iterator]();
+  let i = 0;
+  for (;;) {
+    const { value: char1 } = iter1.next();
+    const { value: char2 } = iter2.next();
+    if (char1 !== char2 || char1 === undefined) {
+      return str1.substring(0, i);
+    }
+    i += char1.length;
+  }
```

Comment:
I’m sure you’re cognizant of the discrepancy here between iterating code points and indexing code units, but that might not be obvious under maintenance of these tests. Anything that compensates for that would be great, albeit accumulating common code points by concatenation or documenting a limitation to correctness under strings where code unit and code point are the same, but if documenting, consider using the dumber algorithm that co-iterates code units to be consistent with the substring call.